### DISCUSSION TITLE
Js macro can't return `false`

### DISCUSSION THREAD

**Discussion by fish-404** [TiddlyWiki/TiddlyWiki5#6773] [Q&A] [answered]
I write a Js macro for some checking. Initially, I want to let it return a `bool` value.  I find it return nothing when the marco return `false`, but it will show text "true" when the macro return `true`. 

I am not sure about if is my macro's mistake. Then I write a simplest Js macro just accept a value then return `bool`, like: 

 ``` js
exports.name = "testBool";

exports.params = [
    {name: "testPara"}
];

/*
Run the macro
*/
exports.run = function(testPara) {
    if (testPara === "true")
        return true;
    else 
        return false;
};
```
If I call the macro like `<<testBool true>>` in tiddler, it will show `true`, but if I call the macro like `<<testBool false>>`, it show nothing. I also try to return `0` or `1`, it seems have the same behavior. 

Finally, I take string `true` and  `false` to return instead. But I wonder if there is something special to have this limit ?

**Comment by Jermolene** [comment, marked as answer]
Hi @fish-404 JavaScript macros in TiddlyWiki must return a string result; Boolean results are not supported.

**Reply by fish-404** [reply]
Thank you. I read the TiddlyWiki Dev document again, find I missing the `run` section point: 
> The run function should return the string value of the macro.

But is it possible to make some hints or forbid `true` result shown when macro called. 
If someone write a JavaScript macro and use other types to return, some result can be shown, and others can't be shown, it seems a little confused. Like I return boolean result incorrectly, the `true` result will be shown, but `false` result won't be shown. It's possible to give a hint or forbid `true` be shown ?

**Reply by Jermolene** [reply]
The result of the macro function is coerced to a string with `.toString()`, so macros that return the wrong type do not cause a failure.
